src/vs/workbench/browser/layout.ts

```diff
@@ -257,6 +264,12 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 		if ((isWindows || isLinux || isWeb) && getTitleBarStyle(this.configurationService, this.environmentService) === 'custom') {
 			this._register(this.titleService.onMenubarVisibilityChange(visible => this.onMenubarToggled(visible)));
 		}
+
+		// Theme changes
+		this._register(this.themeService.onThemeChange(() => this.setWindowBorder()));
```

Comment:
@eamodio I would try to align with what our `Themable` class does (which unfortunately we cannot extend here) and introduce a method `updateStyles()` which then calls `this.setWindowBorder()`.

Reply:
Will do